src/metaopt/io/database/abstractdb.py

```diff
@@ -0,0 +1,56 @@
+import six
+import logging
+
+from metaopt.utils import AbstractSingletonType
+from abc import abstractmethod
+
+@six.add_metaclass(AbstractSingletonType)
+class AbstractDB(object):
+    def __init__(self, dbhost='localhost', dbname='metaopt',
+                 dbport=None, username=None,  password=None):
+        self.dbhost = str(dbhost)
+        self.dbname = str(dbname)
+        self.username = str(username)
+        self.password = str(password)
+        self.dbport = str(dbport)
+
+        self.conn = self.initiate_connection()
+        self.db = None
+
+        self.logger = logging.getLogger('database')
+        self.fh = logging.FileHandler('database.log')
+        formatter = logging.Formatter('%(asctime)s - %(name)s -'
+                                      '%(levelname)s - %(message)s')
+        self.fh.setFormatter(formatter)
+        self.logger.addHandler(self.fh)
```

Comment:
Because it will be used if we would want to have a PostgreSQL or MySQL class down the line

Reply:
They will change the file handler?

Reply:
Shouldn't one prefer just one type of database, so we just need one type of `database.log` file 
 rather than `mongodb-database.log` thus one `database` filehandler? What do you think? :)

Reply:
Yes, but then I still don't see why set self.fh. It seems to me that they can all use self.logger just fine without messing with self.fh.